mpmath/matrices/matrices.py

```diff
@@ -394,9 +394,11 @@ def tolist(self):
     def __repr__(self):
         if self.ctx.pretty:
             return self.__str__()
-        s = 'matrix(\n'
-        s += self._toliststr(avoid_type=True) + ')'
-        return s
+
+        if self.__rows == 0 or self.__cols == 0:
+            return f'matrix({self.__rows}, {self.__cols})'
+
+        return f'matrix(\n{self._toliststr(avoid_type=True)})'
```

Comment:
Why not fix instead the ``_toliststr()`` helper and the matrix constructor?

```pycon
>>> mp.pretty = True
>>> np.array([[]])
array([], shape=(1, 0), dtype=float64)
>>> matrix([[]])  # this probably should be [[]]?
[]
>>> np.array([])
array([], dtype=float64)
>>> _.shape
(0,)
>>> matrix([])  # this doesn't, but it should 
[...]
IndexError: list index out of range
```

And you should add tests.

Reply:
Adjusted the `_toliststr()` helper, to work for the given cases.
But still left  `__nstr__()` which does the pretty printing untouched.
Should it be adjusted aswell?